addons/l10n_cz/tests/test_moves.py

```diff
@@ -54,3 +54,23 @@ def test_cz_bank_rec_no_taxable_supply_date(self):
 
         inv_line = self.env['account.move'].search([('statement_line_id', '=', st_line.id)])
         self.assertNotEqual(inv_line.taxable_supply_date, st_line.date)
+
+    def test_cz_invoice_date_equals_taxable_supply_date(self):
+        """Invoice should keep date = taxable_supply_date for CZ"""
+        self.invoice_a.partner_id = self.env.ref('base.res_partner_12')
```

Comment:
```suggestion
        self.invoice_a.partner_id = self.partner_a
```

Please avoid using demo data inside a test.
Could you also please optimize the test case?